### PR TITLE
Convert mdns Builder classes to use PacketBufferHandle

### PR DESCRIPTION
**Pull request by kpschoedel**
#### Problem

Code should use `PacketBufferHandle` rather than `PacketBuffer`
directly.

#### Summary of Changes

Convert mdns::Minimal::QueryBuilder and mdns::Minimal::ResponseBuilder
to take and hold PacketBufferHandle.

Part of issue #2707 - Figure out a way to express PacketBuffer ownership in the type system

### CODE REVIEW

**Review by saurabhst — APPROVED** [review]

File: examples/minimal-mdns/client.cpp [hunk 1]
```diff
@@ -213,7 +213,7 @@ void BroadcastPacket(mdns::Minimal::ServerBase * server)
     QuerySplitter query;
     query.Split(gOptions.query);
 
-    mdns::Minimal::QueryBuilder builder(buffer.Get_ForNow());
+    mdns::Minimal::QueryBuilder builder(buffer.Retain());
```
Comment (bzbarsky-apple) [inline examples/minimal-mdns/client.cpp:216]: Hmm.  It seems like we could make the builder take a const ref to PacketBufferHandle (and have its member be a const ref) and avoid the extra refcounting  And here we could scope it so it goes out of scope before we do the `BroadcastSend` call...

**Review by andy31415 — APPROVED** [review]

File: examples/minimal-mdns/client.cpp (hunk 1 above)
Comment (kpschoedel) [reply examples/minimal-mdns/client.cpp]: Ah, yes. I probably had TLV's chaining in mind when I converted this one. Fixed.

**Review by bzbarsky-apple — APPROVED** [review]

**Comment by bzbarsky-apple** [comment]
@kpschoedel Merge conflicts?

**Comment by andy31415** [comment]
also #4102 has some conversions to PacketBufferHandle (not sure if the same, but if we could make them similar it may be easier to merge)

**Comment by kpschoedel** [comment]
> also #4102 has some conversions to PacketBufferHandle (not sure if the same, but if we could make them similar it may be easier to merge)

Yes, I think #4102 makes the ResponseBuilder part of this PR redundant, so I'll just remove it.

**Comment by kpschoedel** [comment]
Converted to draft as this now depends on #4102 

Removed ResponseBuilder changes (redundant after #4102) and converted QueryBuilder to use the same interface pattern as ResponseBuilder there.

**Review by rwalker-apple — APPROVED** [review]

**Review by jelderton — APPROVED** [review]

**Review by BroderickCarlin — APPROVED** [review]

**Review by mspang — APPROVED** [review]

**Pull request merged**